### DISCUSSION TITLE
Jamulus -? is not translated. English only

### DISCUSSION THREAD

**Discussion by ghost** [jamulussoftware/jamulus#1660] [Ideas]
**Describe the bug**

Jamulus -? is not translated.
**To Reproduce**

Jamulus -?
**Expected behavior**

Command line help should be in selected language
**Screenshots**

**Operating system**
 All OS

**Version of Jamulus**
 All versions

**Additional context**

**Comment by softins** [comment]
The language selected in the GUI is only for the GUI itself, and is stored in the ini file. At the point where `Jamulus -?` or `Jamulus --help` is executed, the ini file has not been read, and may not even exist.

The only logical item to consult for this would be the environment variable `LANG`. But as a point of comparison, on the Linux systems I use, the command `nano --help` gives the help text in English only, even when I have set `LANG` to other values, such as `fr`, `fr.UTF-8`, `de` or `de.UTF-8`, and `/usr/share/locale/*/LC_MESSAGES/nano.mo` does exist for those languages. Similarly for other commands.

So I don't think Jamulus behaves any worse in this regard.

**Comment by ghost** [comment]
> So I don't think Jamulus behaves any worse in this regard.

If its not a bug, then its not a bug. Maybe close this after a few comments, or days, if that is the right thing to do.

**Comment by hoffie** [comment]
I agree, it's not a bug, because nobody has ever tried to make this work.
Lots of standard Linux tools do have localized `--help` texts (`cat`, `ls`, `grep` and, for me, even `nano` did have it), so I think this is a reasonable request.

It would mean some work for `--help` though and it may trigger the subsequent question if other command line messages would have to be translated as well.

If there's little demand, I'd keep it as-is. If this was helpful for lots of people, we should probably do it.

As you suggest, maybe keep this issue open for some days. It will still be searchable afterwards, so anybody coming across this issue can still voice their opinion later.

**Comment by gilgongo** [comment]
I yield to none in my enthusiasm for making Jamulus available in all languages, but I think if we really want to hit that goal using the resources currently available, `Jamulus -?` should instead point to a page on the web site, which has automatic translation ability.

**Reply by ghost** [reply]
Jamulus uses [ -h -? --help ]
Maybe -? could be online help

**Comment by anttiryt** [comment]
Just no. Making code multilanguage or even multiregion opens a really big Pandora's box. The difference between those is that the latter involves e.g. currencies, time/date output styles, list sort orders.  Making code get it's output from a configuration templates or custom functions or methods. The code easily ends up in a spaghetti no one wants to touch, or pages of code just to do a simple task.

**Reply by henkdegroot** [reply]
What about adding a link in the help to indicate you should access the jamulus website for localised details?

Obvious most users will most likely have found that already.....but just for the few that did not...

**Reply by hoffie** [reply]
> What about adding a link in the help to indicate you should access the jamulus website for localised details?

I think that would be a very good compromise between effort and usability. `For more information and localized help see https://...` at the end of the `--help` text would sound good to me. Maybe someone wants to open a PR for that.

**Reply by ghost** [reply]
> Just no. Making code multilanguage or even multiregion opens a really big Pandora's box. The code easily ends up in a spaghetti no one wants to touch, or pages of code just to do a simple task.

Jamulus is already internationalized. I have never looked into how it is done, but internationalization was figured out a long time ago. I can imagine how spaghetti code, or indexed arrays, or funky c++ code, or even #ifdef's could result from 'roll your own' internationalization.

**Reply by henkdegroot** [reply]
> Maybe someone wants to open a PR for that.

PR #1759 created for this.

**Comment by atsampson** [comment]
It would be easier to translate the help strings for the options if Jamulus used QCommandLineParser, which generates the help message automatically - it might be worth doing that migration before trying to make the existing options code translatable.

**Reply by hoffie** [reply]
> It would be easier to translate the help strings for the options if Jamulus used QCommandLineParser, which generates the help message automatically

I was always wondering why command line parsing was done manually. Would be interesting to see if the code could be converted to `QCommandLineParser` easily, what the effect on length, complexity, compatibility and formatting would be, independent of translations (which I'm still not sure we have to do, especially since @henkdegroot has added the link to the localized docs).
